index.js

```diff
@@ -136,6 +136,19 @@
       return clonedObj
     })
   }
+
+  objectPathImmutable.insert = function (obj, path, value, at) {
+    at = ~~at;
+    return changeImmutable(obj, path, function(clonedObj, finalPath) {
+      var arr = clonedObj[finalPath];
+      if (!isArray(arr))
```

Comment:
We should force `arr` to be an array only if it's not set (NULL or undefined), we should throw an error instead if `arr` is set and it's not an Array, this will avoid silent bugs.